python/ray/experimental/workflow/workflow_storage.py

```diff
@@ -430,18 +436,58 @@ def get_latest_progress(self) -> "StepID":
         return asyncio_run(self._get(self._key_workflow_progress(),
                                      True))["step_id"]
 
+    def _put_object_ref(self, obj_ref: ObjectRef) -> str:
+        @ray.remote
+        def put_helper(identifier: str, obj: Any,
+                       wf_storage: WorkflowStorage) -> str:
+            key = wf_storage._key_obj_id(identifier)
+            loc = asyncio.get_event_loop().run_until_complete(
+                wf_storage._put(key, obj))
+
+            return _load_object_ref, (loc, self._storage)
+
+        identifier = calculate_identifiers([obj_ref]).pop()
+        # TODO (Alex): We should dedupe these puts with the global coordinator.
+        result = ray.get(put_helper.remote(identifier, obj_ref, self))
+        return result
+
     async def _put(self, paths: List[str], data: Any,
-                   is_json: bool = False) -> None:
+                   is_json: bool = False) -> str:
         try:
+            if not is_json:
+                # Setup our custom serializer.
+                output_buffer = io.BytesIO()
+                pickler = cloudpickle.CloudPickler(output_buffer)
+                # NOTE: dispatch table is (unintuitively) a class variable.
+                # That means modifying the class variable will change the
+                # serialization context for `ray.put` too. Therefore, we have
+                # to explicitely make dispatch_table an instance variable.
+                pickler.dispatch_table = pickler.dispatch_table.copy()
+                pickler.dispatch_table[
+                    ray.ObjectRef] = lambda ref: self._put_object_ref(ref)
```

Comment:
I'm a little bit worried about the performance here. Does this mean that we'll store the object ref into storage one by one?

Reply:
oh good point, handled.